### PR TITLE
Update/001

### PR DESCRIPTION
**Pull request by damianryann**
### **In this PR:**

- Completely overhauled props and prop structure within the component ensuring maximum flexibility.
- Typed all props ensuring the component is type safe.
- Documented all props for JSDocs (future implementation).
- Extended SCSS class.

### CODE REVIEW

File: src/styles/index.scss [hunk 1]
```diff
@@ -36,7 +50,7 @@
 // @import '~bootstrap/scss/badge';
 @import '~bootstrap/scss/alert';
 // @import '~bootstrap/scss/progress';
-// @import '~bootstrap/scss/list-group';
+@import '~bootstrap/scss/list-group';
```
Comment (PenguinOfWar) [inline src/styles/index.scss:53]: Just as a thought, for something like this to be useful it probably shouldn't be dependent on the entire BS style library. 

Two things come to mind:

1) style it without bootstrap?
2) an optional bootstrap mode?

**Review by PenguinOfWar — COMMENTED** [review]
Looks good, test coverage would be a good thing to add.

File: src/styles/index.scss (hunk 1 above)
Comment (damianryann) [reply src/styles/index.scss:53]: @PenguinOfWar Good suggestion - I'm happy to style without bootstrap or / copy the bootstrap css keeping the names and properties of the classes so IF someone were to use bootstrap they could modify the classes in their own sass. Removes the dependancy on it and makes it work the same way. What do you think?

**Comment by damianryann** [comment]
> Looks good, test coverage would be a good thing to add.

I plan to add this, will do when I have finalised the modal look.

File: src/styles/index.scss (hunk 1 above)
Comment (PenguinOfWar) [reply src/styles/index.scss:53]: I would give it generic classes that can be styled with custom CSS, and maybe the option to enable default bootstrap classes. Adding overrides is also pretty doable but may be too much if you're already adding your own, sensible, generic classes.

**Pull request merged**